### PR TITLE
LPD-37874 SF

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/components/PaymentMethodModal.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+							<AppDetails
+								backToList={setStep}
+								product={selectedApp}
+							/>
+						)}
+					</ClayModal.Body>
+				</ClayModal>
+			)}
+		</>
+	);
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/components/PaymentMethodModal.tsx:78]:
```suggestion
      if (!open) {
          return null;
      }

	return (
		<ClayModal
			center
			className="clay-modal-refector"
			observer={observer}
			size="full-screen"
		>
			<ClayModal.Header>
				{Liferay.Language.get('add-from-marketplace')}
			</ClayModal.Header>

			<ClayModal.Body className="m-0 p-0">
				{step === PAYMENT_VIEW.list && (
					<InstallPaymentMethodModalBody
						loading={loading}
						pagination={pagination}
						products={items}
						searchQuery={pagination.searchQuery}
						setSearchQuery={pagination.setSearchQuery}
						setSelectedApp={setSelectedApp}
						setStep={setStep}
					/>
				)}

				{step === PAYMENT_VIEW.details && selectedApp && (
					<AppDetails
						backToList={setStep}
						product={selectedApp}
					/>
				)}
			</ClayModal.Body>
		</ClayModal>
	);
```

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+		{
+			label: 16,
+		},
+		{
+			label: 24,
+		},
+		{
+			label: 32,
+		},
+	];
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:44]: Move outside component

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useCallback, useEffect, useState} from 'react';
+
+import {APIResponse, Channel, Product} from '../types/types';
+
+type Sort = 'asc' | 'desc';
+
+const sortIcon = {
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:12]:
```suggestion
const SORT_ICON = {
```

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+	const getCategory = useCallback(
+		async (
+			options?: RequestInit
+		): Promise<APIResponse<any> | undefined | void> => {
+			setLoading(true);
+
+			try {
+				const url = new URL(
+					`https://marketplace-uat.liferay.com/o/headless-admin-taxonomy/v1.0/taxonomy-vocabularies/${11730462}/taxonomy-categories`
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:54]:
```suggestion
					`https://marketplace-uat.liferay.com/o/headless-admin-taxonomy/v1.0/taxonomy-vocabularies/${ADD_A_VARIABLE_HERE_WITH_THIS_VALUE}/taxonomy-categories`
```

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+				if (data?.items?.length) {
+					return data;
+				}
+			}
+			catch (error) {
+				console.error('Failed to fetch channel ID:', error);
+			}
+			finally {
+				setLoading(false);
+			}
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:81]: No need for finally

just add this at the end of the function

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+					'https://marketplace-uat.liferay.com/o/headless-commerce-delivery-catalog/v1.0/channels'
+				);
+
+				url.searchParams.append('search', `name eq '${channelName}'`);
+
+				const response = await Liferay.Util.fetch(url.toString(), {
+					headers: {
+						...options?.headers,
+						'Content-Type': 'application/json',
+						'x-csrf-token': Liferay.authToken,
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:142]: Not the same Liferay to use this token

```suggestion
```

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/style/appDetails.scss [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+.gallery-image.selected {
+	border: 3px solid #0b5fff;
+}
+
+.modal-content {
+	background-color: #f7f7f8;
+	overflow-y: auto;
+	// max-width: var(--container-max-xl, 1248px);
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/style/appDetails.scss:155]: remove comment

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+
+		return () => clearTimeout(handler);
+	}, [searchQuery]);
+
+	useEffect(() => {
+		(async () => {
+
+			// console.log(await getCategory())
+			// console.log(await getVocabulary())
+
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:223]: delete comments

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+		[channelId, debouncedSearchQuery, delta, deltas, fetchChannelId, sort]
+	);
+
+	useEffect(() => {
+		const handler = setTimeout(() => {
+			setDebouncedSearchQuery(searchQuery);
+		}, 1000);
+
+		return () => clearTimeout(handler);
+	}, [searchQuery]);
```
Comment (kevenleone) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts:216]: Create or use a debounce hook, search for it in the portal or copy from Marketplace

**Review by kevenleone — CHANGES_REQUESTED** [review]
Good job @welitonpe I see a lot of work in here.

Please fix the points I've suggested and we'll do another sync later

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Pull request closed without merging**